packages/gestalt/src/IconButton.js

```diff
@@ -87,9 +107,75 @@ const IconButtonWithForwardRef: React$AbstractComponent<
   const classes = classnames(styles.button, touchableStyles.tapTransition, {
     [styles.disabled]: disabled,
     [styles.enabled]: !disabled,
-    [touchableStyles.tapCompress]: !disabled && isTapping,
+    [touchableStyles.tapCompress]:
+      props.role !== 'link' && !disabled && isTapping,
   });
 
+  function handleClick(event) {
+    if (onClick) {
+      onClick({ event });
+    }
+  }
+
+  function handleLinkClick({ event }) {
+    handleClick(event);
+  }
+
+  const renderPogComponent = (selected?: boolean): Node => {
+    return (
+      <Pog
+        active={!disabled && isActive}
+        bgColor={bgColor}
+        dangerouslySetSvgPath={dangerouslySetSvgPath}
+        focused={!disabled && isFocusVisible && isFocused}
+        hovered={!disabled && isHovered}
+        icon={icon}
+        iconColor={iconColor}
+        padding={padding}
+        selected={selected}
+        size={size}
+      />
+    );
+  };
+
+  if (props.role === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        accessibilityLabel={accessibilityLabel}
+        disabled={disabled}
+        inline
+        href={href}
+        onClick={handleLinkClick}
+        ref={innerRef}
+        rel={rel}
+        rounding="circle"
+        tapStyle={disabled ? undefined : 'compress'}
+        target={target}
+      >
+        <div
```

Comment:
@AlbertCarreras setting the classes on an inner div breaks the focus styles, e.g. the icon background does not become darker when focussed.

For `role="button"`
![image](https://user-images.githubusercontent.com/127199/92254146-47bf4580-ee85-11ea-9c02-2c8c4afff55d.png)


For `role="link"`
![image](https://user-images.githubusercontent.com/127199/92254164-50178080-ee85-11ea-9242-24b04df99908.png)

We have the same issue with `<Button />`, so we might have to move away from using `<Link />` and use a regular `<a />` instead. And I would recommend we focus on getting `<Button />` right first.

Reply:
DONE